### PR TITLE
Changes to better support Python 3 for st2client and st2api

### PR DESCRIPTION
**Pull request by tonybaloney**
Makes a small change to the logging module, which patches a builtin method in the logging module. This method signature has changed in Python 3 to accept 1 argument, since st2 doesn't care about that value this just adds it as optional.

Also,

Validated that `python3 -m nose` works in st2client now. Also, tested running `python3 -m nose` under st2client and all the tests now pass.

st2api still needs work

### CODE REVIEW

**Comment by tonybaloney** [comment]
Finished on this PR

**Comment by tonybaloney** [comment]
Current output from st2api integration tests:

```
2018-01-29 15:51:41,662 ERROR [-] Failed registering internal trigger: {'name': 'st2.generic.actiontrigger', 'pack': 'core', 'description': 'Trigger encapsulating the completion of an action execution.', 'payload_schema': {'type': 'object', 'properties': {'execution_id': {}, 'status': {}, 'start_timestamp': {}, 'action_name': {}, 'action_ref': {}, 'runner_ref': {}, 'parameters': {}, 'result': {}}}}.
Traceback (most recent call last):
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/triggers.py", line 91, in register_internal_trigger_types
    trigger_definition=trigger_definition)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/triggers.py", line 55, in _register_internal_trigger_type
    trigger_db = create_shadow_trigger(trigger_type_db)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/services/triggers.py", line 334, in create_shadow_trigger
    return create_or_update_trigger_db(trigger)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/services/triggers.py", line 230, in create_or_update_trigger_db
    trigger_db = TriggerAPI.to_model(trigger_api)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/models/api/trigger.py", line 153, in to_model
    parameters=parameters)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/models/db/trigger.py", line 98, in __init__
    self.uid = self.get_uid()
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/models/db/trigger.py", line 109, in get_uid
    parameters = hashlib.md5(parameters).hexdigest()
TypeError: Unicode-objects must be encoded before hashing
[2018-01-29 15:51:41 +1100] [32112] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/arbiter.py", line 578, in spawn_worker
    worker.init_process()
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/workers/geventlet.py", line 103, in init_process
    super(EventletWorker, self).init_process()
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/workers/base.py", line 135, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/Users/anthonyshaw/repo/st2/virtualenv/lib/python3.6/site-packages/gunicorn/util.py", line 352, in import_app
    __import__(module)
  File "/Users/anthonyshaw/repo/st2/st2api/st2api/wsgi.py", line 25, in <module>
    application = app.setup_app(config)
  File "/Users/anthonyshaw/repo/st2/st2api/st2api/app.py", line 54, in setup_app
    config_args=config.get('config_args', None))
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/service_setup.py", line 123, in setup
    triggers.register_internal_trigger_types()
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/triggers.py", line 91, in register_internal_trigger_types
    trigger_definition=trigger_definition)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/triggers.py", line 55, in _register_internal_trigger_type
    trigger_db = create_shadow_trigger(trigger_type_db)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/services/triggers.py", line 334, in create_shadow_trigger
    return create_or_update_trigger_db(trigger)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/services/triggers.py", line 230, in create_or_update_trigger_db
    trigger_db = TriggerAPI.to_model(trigger_api)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/models/api/trigger.py", line 153, in to_model
    parameters=parameters)
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/models/db/trigger.py", line 98, in __init__
    self.uid = self.get_uid()
  File "/Users/anthonyshaw/repo/st2/st2common/st2common/models/db/trigger.py", line 109, in get_uid
    parameters = hashlib.md5(parameters).hexdigest()
TypeError: Unicode-objects must be encoded before hashing
```

File: st2api/tests/unit/controllers/v1/test_auth_api_keys.py [hunk 1]
```diff
@@ -13,11 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import httplib
-
 import mock
 from oslo_config import cfg
 
+from six.moves import http_client
```
Comment (Kami) [inline st2api/tests/unit/controllers/v1/test_auth_api_keys.py]: Please use consistent import order:

1. System level imports
2. External imports (such as six)
3. Local project imports

Each of the section separated by a blank line.

**Comment by Kami** [comment]
LGTM.

I'm a bit confused, I added a check which runs st2client tests under Python 3 after first set of changes and it passed - https://travis-ci.org/StackStorm/st2/jobs/334606965

Is there something else going on? Are we missing some test coverage?

**Comment by tonybaloney** [comment]
@Kami not sure if it was my virtualenv in particular, but `python3` was a symlink in `virtualenv/bin` to `python`, which is just a copy of the `python2.7` binary it copies when the script is run.

This is starting to run both the integration tests AND to actually run things like st2api under Python 3. I caught a few issues, like the httplib names along the way. Also, some cases where tests were mocking out string literals, which is fine in Python 2, but the API actually returns a byte string so the test wasn't an accurate one

**Comment by Kami** [comment]
Yeah probably just your local environment, because on Travis we explicitly use Python 3 image which only has Python 3 available (and tests were failing until I fixed a bunch of issues to get them to pass).

Do st2api tests already pass under Python 3? If they do, we should also add a check to Travis.

**Comment by tonybaloney** [comment]
The remaining issues were related to my local environment. I started running st2 CLI under Python 3 and then running the API under it. I was on a plane at 35,000ft so not everything could be downloaded. Can add the check to this PR and we can see what it says!

**Comment by Kami** [comment]
Thanks.

I will go ahead, merge it into master and follow the build through to make sure everything is in order (we still don't run end to end tests on user PRs).

**Pull request merged**